README.md

```diff
@@ -104,6 +113,45 @@ These tags always match the following schema: [version]-[identifier].[bump]
 | 1.2.3          | 2.0.0-pre.1       | minor   | no            | N/A           | 1.3.0          |
 | 1.3.0          | 2.0.0-pre.1       | minor   | yes           | filter        | 1.4.0-filter.0 |
 
+### Pre-release Promotion
+
+This feature is use for promoting a pre-release to an offical release that is intended to go out
+into production. In order use this feature will you have to provide `tag-release` with the
+`--promote` as a command-line flag.
+
+```
+Usage:
+   $ tag-release --promote
+
+Example:
+
+   $ tag-release --promote
+   ? Select the pre-release you wish to promote: (Use arrow keys)
+   > v1.1.1-blah.0
+     v2.0.0-another.1
+     v3.0.0-this.5
+```
+
+After selecting the tag you wish to promote it will attempt remove all pre-release commits from history
+and rebase it with master. If conflicts arise you will be promoted to fix the conflicts and then continue
+with the promotion process by running `tag-release --continue`. This cycle will continue until all conflicts
+are resolve and then it will flow as a normal release from that point on.
+
+The `--promote` feature also allows for an optional command-line argument of a tag in the follow form:
+
+```
+Usage:
```

Comment:
This feels a little bit confusing having the Usage and Example. I understand you want to show that the `v` is optional. Maybe remove the Usage section and have two examples (like you do for the Usage section). 

Then either have one of the following...

* a comment at the end of the 2nd example saying that the `v` is optional
`$ tag-release --promote 1.1.1-feature.2 # the "v" is optional`
* update the paragraph above the examples saying the `v` is optional
> The `--promote` feature also allows for an optional command-line argument (`v` is optional) of a tag in the follow form: